Don't ignore draw errors

IceSentry opened the pull request:
# Objective

- It's possible to have errors in a draw command, but these errors are ignored

## Solution

- Return a result with the error

## Changelog

Renamed `RenderCommandResult::Failure` to `RenderCommandResult::Skip`
Added a `reason` string parameter to `RenderCommandResult::Failure`

## Migration Guide
If you were using `RenderCommandResult::Failure` to just ignore an error and retry later, use `RenderCommandResult::Skip` instead.

This wasn't intentional, but this PR should also help with https://github.com/bevyengine/bevy/issues/12660 since we can turn a few unwraps into error messages now.

jgayfer reviewed (approved):
Always nice to have descriptive errors :+1:

IceSentry commented:
Looks like I should use expect here and just log the errors. I'll update my PR soon.

IceSentry commented:
Okay, so this generates a bunch of errors now, but it just generates a bunch RenderCommandFailure. So I'll need to also bubble up what those failures are.

IceSentry commented:
I added a skip variant to `RenderCommandResult` because that's how it's used every where. I also added a new Failure variant with a message so we can log real failures and not just expected situations to skip.

mockersf commented on crates/bevy_pbr/src/render/mesh.rs:
```diff
@@ -2013,7 +2013,7 @@ impl<P: PhaseItem, const I: usize> RenderCommand<P> for SetMeshBindGroup<I> {
                 This is a bevy bug! Please open an issue."
             );
-            return RenderCommandResult::Failure;
+            return RenderCommandResult::Skip;
```
should this one be a failure?

IceSentry replied:
Probably, I just did a batch rename of every `Failure` to `Skip` to keep the current behaviour

IceSentry replied:
done

mockersf commented:
> Looks like I should use expect here and just log the errors. I'll update my PR soon.

you kept 3 `expect`, is it... expected?

IceSentry commented:
> you kept 3 expect, is it... expected?

Yes and no, it's possible I missed one, but also, I want this PR to mostly keep the previous behaviour and we can start slowly finding places that should use `Failure` instead of panicking.

alice-i-cecile commented:
@IceSentry let me know once merge conflicts are resolved so I can merge?
